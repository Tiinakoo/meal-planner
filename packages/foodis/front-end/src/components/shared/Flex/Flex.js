import React from "react";
import styles from "./Flex.module.scss";
import classNames from "classnames";

const Flex = ({
  className,
  children,
  centered,
  centeredVertically = centered,
  centeredHorizontally = centered,
  alignTop,
  alignBottom,
  alignBaseline,
  alignRight,
  spaceBetween,
  wrapContent,
  pileVertically,
  pileVerticallyTo,
  occupyVerticalSpace,
  stretchToSameHeight,
  reverseDirection,
  ...props
}) => {
  const classNameParameters = [
    styles.flex,
    {
      [styles.flex__centeredVertically]: centeredVertically,
      [styles.flex__centeredHorizontally]: centeredHorizontally,
      [styles.flex__alignTop]: alignTop,
      [styles.flex__alignBottom]: alignBottom,
      [styles.flex__alignBaseline]: alignBaseline,
      [styles.flex__alignRight]: alignRight,
      [styles.flex__spaceBetween]: spaceBetween,
      [styles.flex__wrapContent]: wrapContent,
      [styles.flex__stretchToSameHeight]: stretchToSameHeight,
      [styles.flex__pileVerticallyToXsScreen]: pileVerticallyTo === "xs",
      [styles.flex__pileVerticallyToSmScreen]: pileVerticallyTo === "sm",
      [styles.flex__pileVerticallyToMdScreen]: pileVerticallyTo === "md",
      [styles.flex__pileVerticallyToLgScreen]: pileVerticallyTo === "lg",
      [styles.flex__pileVertically]: pileVertically,
      [styles.flex__occupyVerticalSpace]: occupyVerticalSpace,
      [styles.flex__reverseDirection]: reverseDirection,
    },
    className,
  ];

  return (
    <div className={classNames(classNameParameters)} {...props}>
      {children}
    </div>
  );
};

export default Flex;
