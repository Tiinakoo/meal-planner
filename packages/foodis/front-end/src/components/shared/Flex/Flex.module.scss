@import '../../../styles/mixins';

.flex {
  display: flex;
  flex-direction: row;

  &__occupyVerticalSpace {
    height: 100%;
  }

  &__centeredVertically {
    align-items: center;
  }

  &__centeredHorizontally {
    justify-content: center;
  }

  &__alignTop {
    align-items: flex-start;
  }

  &__alignBottom {
    align-items: flex-end;
  }

  &__alignBaseline {
    align-items: baseline;
  }

  &__alignRight {
    justify-content: flex-end;
  }

  &__spaceBetween {
    justify-content: space-between;
  }

  &__wrapContent {
    flex-wrap: wrap;
  }

  &__stretchToSameHeight {
    align-items: stretch;
  }

  &__reverseDirection {
    flex-direction: row-reverse;
  }

  &__pileVerticallyToXsScreen {
    @include xs {
      flex-direction: column;
    }
  }

  &__pileVerticallyToSmScreen {
    @include sm {
      flex-direction: column;
    }
  }

  &__pileVerticallyToMdScreen {
    @include md {
      flex-direction: column;
    }
  }

  &__pileVerticallyToLgScreen {
    @include lg {
      flex-direction: column;
    }
  }

  &__pileVertically {
    flex-direction: column;
  }

  &_flexItem {
    min-width: 1px;
    max-width: 100%;
    flex-grow: 1;
    flex-shrink: 1;
    flex-basis: 0%;
  }

  &__pileVerticallyToXsScreen > &_flexItem {
    @include xs {
      width: 100%;
      flex: none;
    }
  }

  &__pileVerticallyToSmScreen > &_flexItem {
    @include sm{
      width: 100%;
      flex: none;
    }
  }

  &__pileVerticallyToMdScreen > &_flexItem {
    @include md {
      width: 100%;
      flex: none;
    }
  }

  &__pileVerticallyToLgScreen > &_flexItem {
    @include lg {
      width: 100%;
      flex: none;
    }
  }
}
