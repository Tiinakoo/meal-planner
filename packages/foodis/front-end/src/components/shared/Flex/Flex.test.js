import React from "react";

import { mount } from "enzyme";
import Flex from "./Flex";

describe("Flex", () => {
  it("renders", () => {
    const component = mount(<Flex />);

    expect(component).toMatchSnapshot();
  });

  it("renders with alignRight", () => {
    const component = mount(<Flex alignRight />);

    expect(component).toMatchSnapshot();
  });
});
