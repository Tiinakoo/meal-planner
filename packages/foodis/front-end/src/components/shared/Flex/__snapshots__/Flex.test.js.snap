// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders 1`] = `
<Flex>
  <div
    className=""
  />
</Flex>
`;

exports[`Flex renders with alignRight 1`] = `
<Flex
  alignRight={true}
>
  <div
    className=""
  />
</Flex>
`;
